### DISCUSSION TITLE
iOS Extension using MAUI?

### DISCUSSION THREAD

**Discussion by huyvuskedulo** [dotnet/maui#20503] [Q&A]
We're in a process migrating from Xamarin -> MAUI

One of the biggest challenge i'm facing is we developed a Share extension using Xamarin.Forms by following this guidance: https://learn.microsoft.com/en-us/xamarin/ios/platform/extensions-with-xamarinforms.

Do we have a way to build the Share extension by using MAUI Xaml yet? As far as my research, the MAUI framework is tied heavily to AppDelegate and an extension in iOS does not have such AppDelegate like that. And i can't find nay guidance for this.

**Comment by vitorelli** [comment]
Did you try MAUI behaviors?

https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/behaviors?view=net-maui-8.0

**Reply by huyvuskedulo** [reply]
Not sure what you really mean.

The discussion is about iOS Extension able to use MAUI to build the UI, extension like "Share" service, or Push notification extension

**Reply by vitorelli** [reply]
Sorry for misunderstanding you.
About your question the answer is Yes. You need to use MAUI essentials.

https://github.com/dotnet/maui/blob/848202887c7a35d1250c4cb518e899ec65347545/src/Essentials/samples/Samples/ViewModel/ShareViewModel.cs#L142

**Reply by vitorelli** [reply]
For push notifications please check: https://shinylib.net/client/push/

**Reply by RobTF** [reply]
Unfortunately you guys aren't answering the question. OP wants to create a share extension with a custom action related to their app. This isn't the same as simply triggering the share UI from your app.

@huyvuskedulo I believe you need to use the Xamarin templates. I can't find the Github item where this is discussed but I'm sure that's the case.

**Comment by AmanGoyal009** [comment]
I tried to create an Shared Extension in MAUI for IOS. 
I am able to see my application in the Shared Extension but when I tap on the app icon then my app is not opening. And none of the methods inside ShareViewController class are calling. 

So, I can't proceed further. 

My Storyboard is also linked properly with SharedViewController class and added NSExtension in Info.plist file. 

I u guys succeeded then please suggest some solution.
